src/ReturnTypeProvider/ModelReturnTypeProvider.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types=1);
+
+namespace Psalm\LaravelPlugin;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
+use PhpParser\Node\Expr\New_;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Name;
+use Psalm\CodeLocation;
+use Psalm\Context;
+use Psalm\Internal\MethodIdentifier;
+use Psalm\Plugin\Hook\MethodExistenceProviderInterface;
+use Psalm\Plugin\Hook\MethodReturnTypeProviderInterface;
+use Psalm\StatementsSource;
+use Psalm\Type;
+use Psalm\Type\Atomic\TGenericObject;
+use Psalm\Type\Atomic\TNamedObject;
+use Psalm\Type\Atomic\TTemplateParam;
+use Psalm\Type\Union;
+
+final class ModelReturnTypeProvider implements MethodReturnTypeProviderInterface
```

Comment:
@muglug , @mr-feek , do you think implementation of such provider is better than fix #3219 ?

Reply:
I do not think this is a good implementation approach -- I'm trying to solve by templating. Just using  this draft as a brainstorming area..

Reply:
This was how we solved it over at larastan, but proxying for return types is much easier over there. If we can solve via templating, I think we should strive for that appraoch!

Reply:
Correct me if I'm wrong but it's easy task: `@return static&Builder<static>` for several model methods. WDYT?

Reply:
Yeah, I've been debating on just hardcoding `Builder` vs figuring out the actual type of builder etc. Feel free to make a PR if you have ideas!

Reply:
I think the easiest is to just copy/paste the builder methods and put them in the model stub. I've been messing with rewriting the docblocks that ide helper generates to make them correct.